### ISSUE TITLE
Implement javax.xml.validation support

### ISSUE THREAD

**Issue by rosslamont** [phax/ph-schematron#46]
It would be great if standard xml validation techniques could be used.  You'd need some extensions to enable phase support.

Happy to help if you point me in the right direction.  I'm working on https://github.com/rosslamont/jxvc which is primarily about xml-model support but I need a standards based schematron validator to round it out.

**Comment by phax** [comment]
* There is support for phase, as long as you use the `SchematronResourceSCH` - the XSLT based version already has a phase applied
* What exactly do you need?
* I already have a validation stack that combines XSD and SCH - https://github.com/phax/peppol-validation-engine

**Comment by rosslamont** [comment]
I wasn't clear at all in my original request.  I was looking for a JAXP validation engine, i.e. implementations of the classes in javax.xml.validation.  Perhaps I'm not looking hard enough, but it's not obvious where/if you implement those.  Also, your validator seems (without spending too much time looking at it) to not be a streaming (SAX style) validator.  I know Schematron probably doesn't fit too well with a streaming validator, but in order to fit the javax.xml.validation model it would have to stream.

To be more exact, I need a pure streaming Schematron validator that does not do any other sort of validation (e.g. XSD).  The xml-model spec which my validator implements covers situations where you might want to do both XSD and Schematron.

By all means, let me know if I'm missing something and need to look into your stuff further.

**Comment by phax** [comment]
Thanks for the clarification.
First of all: Schematron is not suitable for streaming, except you are using a special language for the rules called STX - which is a subset of XSLT suitable for streaming. But XSLT per se is not suitable for streaming.

In general my solution is not necessarily using JAXP for parsing.
Anyway the XML document to be validated must be present as DOM node.

As I stumbled upon the same issue, I started looking a bit into STX but didn't came much further than https://github.com/phax/ph-stx which is a version of Joost which I want to base on another grammar etc.

And concerning JAXB validation I may point you to my ph-jaxb subproject of https://github.com/phax/ph-commons - it provides an abstraction layer for all kind of JAXB usage (read and write).
